src/transport_performance/gtfs/gtfs_utils.py

```diff
@@ -60,3 +63,88 @@ def bbox_filter_gtfs(
     print(f"Filtered feed written to {out_pth}.")
 
     return None
+
+
+# NOTE: Possibly move to a more generalised utils file
+def convert_pandas_to_plotly(
+    df: pd.DataFrame, return_html=False, scheme: str = "dsc"
+) -> go.Figure:
+    """Convert a pandas dataframe to a visual plotly figure.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        A pandas dataframe to convert to plotly
+        (single index only)
+    return_html : bool, optional
+        Whether or not to return the html element,
+          by default False
+    scheme : str, optional
+        The colour scheme of the dataframe,
+        by default "dsc"
+
+    Returns
+    -------
+    go.Figure
+        A plotly figure containing the drawn dataframe
+
+    Raises
+    ------
+    LookupError
+        An error raised if an invalid colour scheme is passed
+    TypeError
+        An error raised if the given pandas dataframe is MultiIndex
+
+    """
+    # pre-defined colour schemes
+    schemes = {
+        "dsc": {
+            "header_fill": "#12436D",
+            "header_font_colour": "white",
+            "cell_fill": "#A285D1",
+            "cell_font_colour": "black",
+            "font_family": "sans-serif",
+            "line_colour": "black",
+        }
+    }
+    # defences
+    _type_defence(df, "df", pd.DataFrame)
+    _type_defence(return_html, "return_html", bool)
+    if scheme not in list(schemes.keys()):
+        raise LookupError(
+            f"{scheme} is not a valid colour scheme."
+            f"Valid colour schemes include {list(schemes.keys())}"
+        )
+    if isinstance(df.columns, pd.MultiIndex):
+        raise TypeError(
+            "Pandas dataframe must have a single index," "not MultiIndex"
+        )
```

Comment:
Passes on df with multi index but raises when columns are multi index. Is this the behaviour you need? If so, make clear in TypeError.

Possibly raise when either index or columns are multi?